### ISSUE TITLE
Events Accessibility audit

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#17660]
## Description
Complete a full accessibility audit of the Events product, including:
- https://www.va.gov/boston-health-care/events
  - https://www.va.gov/boston-health-care/events/66748/
  - https://www.va.gov/boston-health-care/events/65056/
- https://www.va.gov/outreach-and-events/events
  - https://www.va.gov/outreach-and-events/events/64661/
  - https://www.va.gov/outreach-and-events/events/64842/

## Acceptance Criteria

- [x] Accessibility audit has been completed
- [x] Recommended updates and improvements have been documented
- [x] Any high priority/critical updates are brought to the attention of PM/DM/PO
- [x] Design review ticket is created

**Comment by laflannery** [comment]
### Accessibility Updates
- On recurring events, there are multiple duplicate links for “Add to calendar”. These links aren’t unique and a user has no way to differentiate these from one another. 
  - There is [a backlog issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1644) to add aria-label to <va-link> which would handle this so when this is adopted we can use it to support this
- There are heading issues within the event body but this is created with a rich text editor so it’s not easily fixed.
- Paging on listing needs to have proper focus management with results listing. There is already [a ticket for this](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14755).
- The dates filters so not show the error message when the Apply is clicked, these are only show when the user clicks or tabs out of the fields
  - This was an existing issue before the component upgrade

### Updates that should be made to align with upcoming DST changes:

**`<va-icon>` migration**
Planned for deprecation of current icons beginning of May
- Icons on Calendars links on Event detail pages
  - This is contained within the component so it might get auto updated
  - Recurring events have multiple of these links
- Icons on Social links on Event detail pages

**Upcoming Heading/Link guidance**
No timeline yet for when [guidance](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2146) will be formalized
- Should the headings on the listing be unlinked and a separate CTA be added to align with upcoming guidance or is this an exception?

**Components**
- [Action Link](https://design.va.gov/components/link/action)
  - Will need to be updated when ready for use from DST, [planned for mid-April](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2500)
  - Where it’s used:
    - Register link currently uses primary action link
- [Link](https://design.va.gov/components/link/)
  - Directions link because this doesn’t need to open in a new tab anymore

### Other updates/improvements:
- There is an empty <va-link> component
  - It is a link for the “event-fieldLocationTitle”. This is coming from the “Facility Location” field when the “At a VA Facility radio button is selected. This field isn’t available when the event is not at a VA Facility.
  - So this needs to updated to include conditional logic
  - This was not a result of the recent component updates, it was actually an existing issue
- Consider componentizing the side nav
- Consider updating the filters to use just [the month and year option](https://design.va.gov/components/form/date-input#monthyear)
  - It seems clunky to have to choose a specific date. Maybe we have stats to know how often this is used and can make a more educated decision on this?
